src/components/views/rooms/BasicMessageComposer.tsx

```diff
@@ -480,11 +480,22 @@ export default class BasicMessageEditor extends React.Component<IProps, IState>
                 handled = true;
                 break;
         }
+
         if (handled) {
             event.preventDefault();
             event.stopPropagation();
             return;
         }
+        // Always prevent the default if key === Enter. There are multiple reasons for this:
+        // 1) It makes it possible to set the NewLine key binding to a key other than Enter without still triggering a
+        // new line when pressing Enter.
+        // 2) It stops key combos like shift + Enter or alt + Enter to generate NewLines.
+        // 3) It avoids a Chrome 'quirk' which sometimes results in a '\n\n' input on shift+Enter
+        // (https://github.com/vector-im/element-web/issues/17215).
+        if (event.key === Key.ENTER) {
+            event.preventDefault();
+            return;
+        }
```

Comment:
All of this code doesn't appear to be needed anymore, at least not in testing.

Reply:
The issue isn't triggered if `shift + Enter` OR `Enter` is used as key combo to insert a new line. @SimonBrandner is working on customizable key bindings though. For this to work the check is still needed, e.g. if you put new line on something other than `Enter`. For this reason I would leave it in. I updated the outdated comment...

Reply:
@turt2live any opinion on that? Still think it makes sense to keep it in and the issue documented in the code.

Reply:
The PR worked just fine for all combinations when I removed this whole block, so am curious as to what it's trying to do. The comment isn't super revealing into why it's needed specifically, but does accurately transcribe the issue to the code. Our comments generally strive to explain the *why* over the *how*, which means describing both the problem and solution, and why the solution is the way it is - clarifying the block with this format might help understand what this block is doing.

Reply:
? there are three reasons in the comment that explain the _why_ :-) However, I think I can simplify the comment:

"Swallow the Enter key event in case the NewLine key action is mapped to something different than a key combo containing the Enter key."

Would that be clearer?

Reply:
possibly, but I'm still curious as to why the code block doesn't seem to be doing anything.

Reply:
The code block is for when the NewLine action is mapped to somethings else than the Enter key (assuming this will be configurable at some point and somebody actually want to change it...). With the default config the code block is never reached because the key combo is handled earlier.

Reply:
@czeidler, after some thinking, I think we might actually want to handle the send shortcut a little differently than the other shortcuts, so I would suggest removing this code, for now, we can always get it back if we need to in the future

Reply:
which code? not sure what you mean

Reply:
The changes in `KeyBindingDefaults.ts` are sufficient to fix the issue at hand therefore the changes in `BasicMessageComposer.tsx` aren't necessary. If we want a more general fix in the future we can always get the changes to `BasicMessageComposer.tsx` from this PR but for now, we don't need them